modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -1936,7 +1936,10 @@ def str_split(self, pat=None, n=-1, expand=False, regex=None):
     def unique(self, keep="first", ignore_index=True, subset=None):
```

Comment:
@dchigarev when [combining functions](https://github.com/modin-project/modin/pull/7091) `drop_duplicates` and `unique` into one, did you take into account that they return different data types? It seems to me that because of this, using `qc.unique` in its current form is inconvenient. From my side, the best solution seems to be to split this function into several. What do you think?

Reply:
> when https://github.com/modin-project/modin/pull/7091 drop_duplicates and unique into one, did you take into account that they return different data types

Initially, I haven't seen a problem with it, since the result of both `.unique()` and `.drop_duplicates()` was wrapped into a pandas dataframe anyway inside of partition class. I didn't want to add another unique-like qc method in order not to bloat QCs api (which is already quite big and sometimes redundant).

An ideal solution I see is to continue wrapping the results with a pandas dataframe inside of partitions and then make the front-end handle the conversion to the desired output type (for example, `.drop_duplicates()` returns a `DataFrame(result_qc)` and `.unique()` returns `result_qc.to_values()`). Arrow extension arrays seem to round-trip to pandas with no problems:
```python
>>> arr = pd.array([1, 1, None], dtype="int64[pyarrow]")
>>> type(arr)
<class 'pandas.core.arrays.arrow.array.ArrowExtensionArray'>
>>> arr is pd.DataFrame(arr).squeeze().values
True
```